python/ray/_raylet.pyx

```diff
@@ -222,6 +222,53 @@ cdef increase_recursion_limit():
                          new_limit, current_depth))
 
 
+cdef CObjectLocationPtrToDict(CObjectLocation* c_object_location):
+    """A helper function that converts a CObjectLocation to a Python dict.
+
+    Returns:
+        A Dict with following attributes:
+        - primary_node_id:
+            The hex ID of the node who has the primary copy of the object.
+            It could be None if the object is pending resolution.
+        - object_size:
+            The size of data + metadata in bytes.
+        - node_ids:
+            The hex IDs of the nodes that have an in memory copy of this
+            object.
+        - is_spilled:
+            Wether the object has been spilled.
+        - spilled_url:
+            The spilled location, None if not spilled.
+        - spilled_node_id:
+            The hex node ID which spilled the object. None if the object
+            is not spilled, or was spilled to a distributed external
+            storage.
```

Comment:
Can we drop this too (can roll it into node ids). It's also kind of an implementation detail (fs spilling).